layer_fs_wrapper.go

```diff
@@ -0,0 +1,104 @@
+package claircore
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/quay/zlog"
+)
+
+// FilterFS wraps an fs.FS and hides inaccessible files and directories
+// by returning appropriate "not found" errors or filtering them from listings.
+type FilterFS struct {
+	fsys fs.FS
+	ctx  context.Context
+}
+
+// New creates a new FilterFS wrapper around the provided filesystem.
+func NewFilterFS(fsys fs.FS, ctx context.Context) *FilterFS {
```

Comment:
If we keep the Context in the struct, should it come first in this function? It _usually_ comes first, but perhaps having it come second would signal to the caller that we're using it a bit differently here.